asapdiscovery-modeling/asapdiscovery/modeling/modeling.py

```diff
@@ -1,32 +1,184 @@
+    """
+    Prepares a protein for docking. Current implementation uses schema from modeling.schema
+    Automatically logs to a file in the output directory (specified by prep_opts).
+    Contains the following steps:
+    - Load structure (cif or pdb)
+    - Get desired components specified by target.molecule_filter
+    - Align if requested
+    - Mutate Residues using SEQRES
+    - Spruce
+    - Saves either a failed spruced pdb or a complete set of prep outputs
+
+
+    Args:
+    - target (PreppedTarget): the target to prepare
+    - prep_opts (PrepOpts): the options for preparing the target
+
+    Returns:
+    - PreppedTarget: the prepared target
+    """
+    target.output_dir = Path(prep_opts.output_dir / target.output_name)
+    if not target.output_dir.exists():
+        target.output_dir.mkdir(parents=True)
+
+    logger = FileLogger(
+        logname=f"protein_prep_workflow.{target.output_name}",
+        path=str(target.output_dir),
+    ).getLogger()
+
+    # TODO: Add back check to make sure that the target is not already prepared
+    #  I'd like to do something clever with reading the output PreppedTargets file so I'll wait until
+    #  We have the schema finalized.
+
+    logger.info(f"Preparing {target.output_name}")
+
+    # Load structure
+    if Path(target.source.str_fn).suffix == ".pdb":
+        prot = load_openeye_pdb(target.source.str_fn)
+    elif Path(target.source.str_fn).suffix == ".cif":
+        prot = load_openeye_cif1(target.source.str_fn)
+    else:
+        raise NotImplementedError(
+            f"Cannot load structure with extension {Path(target.source.str_fn).suffix}"
+        )
+    logger.info(f"Loaded {target.source.str_fn}")
+
+    # Get desired components
+    logger.info(f"Splitting molecule using {target.molecule_filter}")
+    prot = split_openeye_mol(prot, target.molecule_filter)
+
+    # Align
+    if prep_opts.ref_fn:
+        logger.info(f"Aligning to {prep_opts.ref_fn}")
+        ref = load_openeye_pdb(str(prep_opts.ref_fn))
+        prot, rmsd = superpose_molecule(
+            ref, prot, prep_opts.ref_chain, target.active_site_chain
+        )
+
+    # Mutate Residues
+    if prep_opts.seqres_yaml:
+        logger.info(f"Mutating residues using {prep_opts.seqres_yaml}")
+        with open(prep_opts.seqres_yaml) as f:
+            seqres_dict = yaml.safe_load(f)
+        seqres = seqres_dict["SEQRES"]
+
+        res_list = seqres_to_res_list(seqres)
+
+        prot = mutate_residues(prot, res_list, place_h=True)
+        seqres_list = " ".join(res_list)
+    else:
+        seqres = None
+        seqres_list = None
+
+    # Spruce Protein
+    logger.info("Running OESpruce using spruce_protein")
+    du = spruce_protein(
+        initial_prot=prot,
+        seqres=seqres_list,
+        loop_db=prep_opts.loop_db,
+        return_du=True,
+        site_residue=target.oe_active_site_residue,
+    )
+    if type(du) == oechem.OEGraphMol:
```

Comment:
we discussed this - 
I will split up spruce_protein to always return a GraphMol and then have prep_protein_workflow call a second function that makes design units from that GraphMol. This can be set as an option within the prep_protein_workflow